pkgs/by-name/sm/sm64coopdx/package.nix

```diff
@@ -0,0 +1,130 @@
+{
+  fetchFromGitHub,
+  lib,
+  writeTextFile,
+
+  bash,
+  curl,
+  hexdump,
+  python3,
+  SDL2,
+  stdenv,
+  zlib,
+
+  sm64baserom,
+  enableCoopNet ? true,
+  enableDiscord ? true,
+  enableTextureFix ? true,
+}:
+let
+  libc_hack = writeTextFile {
+    name = "libc-hack";
+    # https://stackoverflow.com/questions/21768542/libc-h-no-such-file-or-directory-when-compiling-nanomsg-pipeline-sample
+    text = ''
+      #include <unistd.h>
+      #include <string.h>
+      #include <pthread.h>
+    '';
+    destination = "/include/libc.h";
+  };
+  baserom = sm64baserom.override {
+    region = "us";
+    showRegionMessage = false;
+  };
+  pname = "sm64coopdx";
+  version = "1.0.3";
```

Comment:
unless there's some benefit in being generic, it's easier to read the package if you just inline `sm64coopdx` instead of making it variable.

for `version`, you can either change `stdenv.mkDerivation { ... }` to `stdenv.mkDerivation (finalAttrs: { version = "1.0.3"; ... })` and refer to `finalAttrs.version`, or use `stdenv.mkDerivation rec { version = "1.0.3"; ... }`.